.pre-commit-config.yaml

```diff
@@ -0,0 +1,53 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-json
+    -   id: check-added-large-files
+-   repo: https://github.com/python-jsonschema/check-jsonschema
+    rev: 0.13.0
+    hooks:
+    # Following only exists on non-EOL Python versions
+    #-   id: check-metaschema
+    #    name: "Check MSR Definition Format"
+    #    files: json_schemas/[^/]+\.schema\.json$
+    -   id: check-jsonschema
+        name: "Check MSR Definition Format"
+        files: IGNOREFORNOW/msr_[^/]+\.json$
+        args: ["--schemafile", "service/json_schemas/msrs.schema.json"]
+-   repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+    -   id: rst-backticks
+    -   id: rst-directive-colons
+    -   id: rst-inline-touching-normal
+-   repo: https://github.com/pycqa/flake8
+    rev: '5.0.4'
+    hooks:
+    -   id: flake8
+        args: ["--max-line-length=120", "--extend-ignore=E261"]
+-   repo: https://github.com/pycqa/bandit
+    rev: '1.7.0'
+    hooks:
+    -   id: bandit
+        # Only gate high-severity issues
+        args: ["-lll"]
+-   repo: https://github.com/codespell-project/codespell
+    rev: 'v2.2.0'
+    hooks:
+    -   id: codespell
+        args: ["--ignore-words-list", "atleast,clos"]
+-   repo: local
+    hooks:
+    -   id: geopm-license
+        name: GEOPM License Checks
+        description: This hook checks whether files have the right license header.
+        pass_filenames: false
+        entry: ./check-precommit-license
+        language: script
+        types: [file]
```

Comment:
We can also run scripts to do any project-specific checks. We want these to be fast things (like checking for license headers), not long things that would disrupt commit workflows (like build-and-UT). Anything else to add?